docs/NNNN-direct-clean-path-layout.md

```diff
@@ -0,0 +1,255 @@
+make sure, that there are no possibly dangerous file- or foldernames, which 
+may oppose with durability.
+
+One Example could be complete filesystems, which come from estates, and are not
+under control of the archivists.
+
+### Caveat
+
+#### Projection
+This extension does  provide injective (one-to-one) projections only if 
+`encodeUTF == true`.  
+If you don't want to use the longish UTF-Code replacements, first check your source to make sure, that there's no chance of 
+different names to be mapped on one target. Software which generates the OCFL 
+structure should raise an error in this case.
+
+#### Object Identifiers
+Besides the caveat below, when using as storage layout root extension, you must 
+be sure, that no object identifier is  contained in another identifier as a 
+prefix path.  
+I.e. https://hdl.handle.net/XXXXX/test and https://hdl.handle.net/XXXXX/test/blah 
+could result to into storage root folders like  
+https/hdl.handle.net/XXXXX/test    
+https/hdl.handle.net/XXXXX/test/blah  
+This is an invalid OCFL storage root hierarchy.
+
+#### Length
+Several filesystems (e.g. Ext2/3/4) have byte restrictions on filename length.
+When using UTF16 (i.e. NTFS) or UTF32 characters in the filesystem, the byte length is double or quad of the character length.
+Using UTF8 characters in filesystems, byte length can only be determined by checking each string for maximum.  
+**Hint:** UTF8 has always less or equal bytes than UTF32 which means, that 
+assuming UTF32 instead of UTF8 for length calculation is safe, but would give you
+only 63 characters on 255 byte restrictions. 
+
+## Parameters
+
+### Summary
+
+* **Name:** `encodeUTF`
+    * **Description:** decides whether "dangerous" characters will be replaced by a 
+      defined replacement string or it's utf code (e.g. =u0020 for blank char) 
+    * **Type:** bool
+    * **Default:** false
+* **Name:** `maxFilenameLen`
+   * **Description:** determines the maximum number of characters within parts of the full pathname separated by `/` (files or folders).   
+     A result with more characters will raise an error.  
+    * **Type:** number
+   * **Constraints:** An integer greater 0
+   * **Default:** 127
+* **Name:** `maxPathnameLen`
+    * **Description:** determines the maximum number of result characters.
+      A result with more characters will raise an error.  
+    * **Type:** number
+    * **Constraints:** An integer greater 0
+    * **Default:** 32000
+* **Name:** `replacementString`
+  * **Description: String which is used to replace non-whitespace characters 
+    which need replacement.  
+    If `encodeUTF == true` only used for replacement of non-UTF8 characters.
+  * **Type:** string
+  * **Default:** "_"
+* **Name:** `whitespaceReplacementString`
+    * **Description:** String which is used to replace whitespaces (https://en.wikipedia.org/wiki/Template:Whitespace_(Unicode)).  
```

Comment:
Use "that" instead of "which" here too.

Is the link supposed to be a definition of whitespace? If so, it should probably be a link like: [whitespaces](https://en.wikipedia.org/wiki/Template:Whitespace_(Unicode))